File: AkkaWithZeroMQ/src/main/scala/org/akka/essentials/zeromq/example4/PullActor2.scala
package org.akka.essentials.zeromq.example4
import akka.zeromq.ZeroMQExtension
import akka.actor.ActorLogging
import akka.zeromq.ZMQMessage
import akka.actor.Actor
import akka.zeromq.SocketType
import akka.zeromq.Connect
import akka.zeromq.Listener

class PullActor2 extends Actor with ActorLogging {
  val pullSocket = ZeroMQExtension(context.system).newSocket(SocketType.Pull, Connect("tcp://127.0.0.1:1234"), Listener(self))

  def receive: Receive = {
    case m: ZMQMessage =>
      var mesg = new String(m.payload(0))
      log.info("Received Message -> {}", mesg)
  }
}

File: AkkaWithZeroMQ/src/main/java/org/akka/essentials/zeromq/example4/PullActor2.java
package org.akka.essentials.zeromq.example4;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.zeromq.Connect;
import akka.zeromq.Listener;
import akka.zeromq.SocketOption;
import akka.zeromq.ZMQMessage;
import akka.zeromq.ZeroMQExtension;

public class PullActor2 extends UntypedActor {

	ActorRef pullSocket = ZeroMQExtension.get(getContext().system())
			.newPullSocket(
					new SocketOption[] { new Connect("tcp://127.0.0.1:1237"),
							new Listener(getSelf()) });
	LoggingAdapter log = Logging.getLogger(getContext().system(), this);

	@Override
	public void onReceive(Object message) throws Exception {

		if (message instanceof ZMQMessage) {
			ZMQMessage m = (ZMQMessage) message;
			String mesg = new String(m.payload(0));
			log.info("Received Message -> {}",mesg);
		}

	}
}
